### PR TITLE
Add support for multi-type unions.

### PR DESCRIPTION
**Pull request by enifsieus**
This commit adds support for multi-union types, while preserving the generation of optional types for unions that are only null + a single type.

### CODE REVIEW

**Comment by enifsieus** [comment]
Greetings - I'm wondering if this is a pull request that anyone is interested in discussing. Thanks!

**Comment by ffinfo** [comment]
Sorry missed out some notifications on this project, will look into this. Sorry for the delay.

File: pyproject.toml [hunk 1]
```diff
@@ -1,6 +1,6 @@
 [tool.poetry]
 name = "pydantic-avro"
-version = "0.4.2"
+version = "0.4.3"
```
Comment (ffinfo) [inline pyproject.toml:3]: please no updates to the versions, this will happen automatically on release

File: tests/test_from_avro.py [hunk 2]
```diff
@@ -228,3 +228,34 @@ def test_enums_reuse():
...
+                                }
+                            ]
+                        }
+                    ]
+                }
+            ]
+        }
+    )
+
+    assert "a_union: Union[None,int,str,ARecord]" in pydantic_code
```
Comment (ffinfo) [inline tests/test_from_avro.py:261]: This is incorrect, should be `a_union: Optional[int,str,ARecord] `

File: tests/test_to_avro.py [hunk 3]
```diff
@@ -76,6 +80,10 @@ class ModelWithAliases(AvroBase):
     field: str = Field(..., alias="Field")
 
 
+class ModelWithUnion(AvroBase):
+    field: Union[None, str, int, NestedModel]
```
Comment (ffinfo) [inline tests/test_to_avro.py:84]: Should be Optional instead of union with None in there

**Review by ffinfo — CHANGES_REQUESTED** [review]

File: pyproject.toml (hunk 1 above)
Comment (ffinfo) [reply pyproject.toml]: with with merge commit

**Comment by ffinfo** [comment]
@enifsieus thanks for this, just merged this, will release this soon

**Pull request merged**